src/plugins/intel_gpu/src/runtime/memory_pool.cpp

```diff
@@ -93,7 +93,7 @@ void memory_pool::release_memory(memory* mem, const size_t& unique_id, primitive
             auto list_itr = list.begin();
 
             while (list_itr != list.end()) {
-                if (list_itr->_memory.get() == mem &&
+                if (list_itr->_memory.get()->buffer_ptr() == mem->buffer_ptr() &&
```

Comment:
@yeonbok  in case of OpenCL buffers both results of buffer_ptr() would be nullptr
Probably it should be `list_itr->_memory.get()->get_internal_params().mem == mem->get_internal_params().mem` check instead

Reply:
Thanks for the comment I will update soon

Reply:
done